frontend/src/test/setup.ts

```diff
@@ -1,9 +1,12 @@
 import '../css/index.css';
 import '@testing-library/jest-dom/vitest';
 import { cleanup } from '@testing-library/react';
-import { afterEach } from 'vitest';
+import { afterEach, vi } from 'vitest';
 import '../i18n';
 
 afterEach(() => {
   cleanup();
+
+  vi.clearAllMocks();
+  vi.restoreAllMocks();
```

Comment:
great idea that we are moving this out to the setup file so it can be consistently run. do avoid duplication, can you please remove `clearAllMocks` from all the other files that are still using it? I see quite a bit